### ISSUE TITLE
Elasticsearch Cluster 5.5.1-rancher1 not deployable

### ISSUE THREAD

**Issue by ckauf** [rancher/community-catalog#756]
Hi *,

when trying to deploy the ES community template 5.5.1-rancher1 (and also 5.4.0-rancher1 and 5.4.2-rancher1)  I get the following error:
```
Error (can't have more than one primary service [es-client es-data] referencing the same sidekick [es-storage]
```
I guess this is the problem in the compose file:
```
es-data:
  [...]
  volumes_from:
    - es-storage

es-client:
  [...]
  volumes_from:
    - es-storage
```
I am running this software stack:

Component|Version
-------------|---------
Rancher       | v1.6.15
Cattle          | v0.183.45
User Interface | v1.6.42
Rancher CLI | v0.6.8
Rancher Compose | v0.12.5

I have another rancher cluster where the same template can be deployed just fine.
Software Versions:

Component|Version
-------------|---------
Rancher | v1.6.14
Cattle | v0.183.37
User Interface | v1.6.37
Rancher CLI | v0.6.7
Rancher Compose | v0.12.5

I've checked the changelog, but could not find any hints what may explain this issue.
Is there anything I can add?

Regards,
Chris

**Comment by krisanalfa** [comment]
Any workaround?

**Comment by ckauf** [comment]
I downgraded my Rancher Cluster to 1.6.14 as described in the section "Important - Upgrade" that can be found [here](https://github.com/rancher/rancher/releases/tag/v1.6.15). Unfortunately I do not know enough about the rancher-compose format to fix this, yet.

**Comment by junkiebev** [comment]
I fixed this by tweaking a few things on the latest version in my company published public catalog: https://github.com/INTLFCStone/Rancher-Catalog

Take a look see @krisanalfa @ckauf

**Comment by joshiegy** [comment]
Same issue for version 6.2.3

**Comment by phynias** [comment]
+1 is there a fix or workaround ??

**Comment by mkoczorowski** [comment]
+1

**Comment by rawmind0** [comment]
Hi all,

issue is related to [#10563](https://github.com/rancher/rancher/issues/10563) and merged PR [#143](https://github.com/rancher/rancher-compose-executor/pull/143) included at rancher v1.6.15. It avoid the use same sidekick for distinct services, like `es-storage` is defined at es-cluster catalog package. 

The solution is redefine storage sidekick `es-storage` to single storage sidekick for every es-cluster service, `es-master-storage`, `es-data-storage` and `es-client-storage`. Created and merged PR https://github.com/rancher/community-catalog/pull/789 that address it.

**Comment by rawmind0** [comment]
Feel free to reopen if still have issues.
